virtcontainers/device/drivers/vfio.go

```diff
@@ -209,8 +209,18 @@ func (device *VFIODevice) Load(ds persistapi.DeviceState) {
 
 // It should implement GetAttachCount() and DeviceID() as api.Device implementation
 // here it shares function from *GenericDevice so we don't need duplicate codes
+
 func getVFIODetails(deviceFileName, iommuDevicesPath string) (deviceBDF, deviceSysfsDev string, vfioDeviceType config.VFIODeviceType, err error) {
-	vfioDeviceType = GetVFIODeviceType(deviceFileName)
+	tokens := strings.Split(deviceFileName, ":")
+	vfioDeviceType = config.VFIODeviceErrorType
+	if len(tokens) == 3 {
+		vfioDeviceType = config.VFIODeviceNormalType
+	} else {
+		tokens = strings.Split(deviceFileName, "-")
+		if len(tokens) == 5 {
```

Comment:
This test isn't exhaustive - what if there aren't 3 or 5 tokens? Error? I think a set of unit tests for this function would be useful:

- https://github.com/kata-containers/tests/blob/master/Unit-Test-Advice.md#table-driven-tests

Reply:
This isn't new logic, it's just been moved here from `GetVFIODeviceType`, so I don't think fixing it is in scope for this patch.

That said, I think it's actually ok.  If neither of the conditions hit, then vfioDeviceType remains as config.VFIODeviceErrorType which gets reported a little below.

Reply:
Ack - thanks for checking.